src/rf/js/src/home/components/modals.js

```diff
@@ -122,30 +124,50 @@ var UploadModal = React.createBackboneClass({
         return fileNameParts[fileNameParts.length - 1];
     },
 
-    s3BucketKeyValid: function(s3BucketKey) {
-        // example valid s3BucketKey: ab-cd.ef/gh1/i-j.k
-        var re = /^([\w-.]+\/)+[\w-]+\.\w+$/;
-        return re.test(s3BucketKey);
+    isTiff: function(fileName) {
```

Comment:
We should definitely coalesce the S3 copy & file upload validation logic in the future. Hopefully, this is Good Enough (:tm:) for now.